plugins/mentions/plugin.js

```diff
@@ -0,0 +1,448 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+'use strict';
+
+( function() {
+
+	CKEDITOR._.mentions = {
+		cache: {}
+	};
+
+	var MARKER = '@',
+		MIN_CHARS = 2,
+		cache = CKEDITOR._.mentions.cache;
+
+	CKEDITOR.plugins.add( 'mentions', {
+		requires: 'autocomplete,textmatch,ajax',
+		instances: [],
+		init: function( editor ) {
+			var self = this;
+
+			editor.on( 'instanceReady', function() {
+				CKEDITOR.tools.array.forEach( editor.config.mentions || [], function( config ) {
+					self.instances.push( new Mentions( editor, config ) );
+				} );
+			} );
+		}
+	} );
+
+	/**
+	 * Mentions plugin allows you to type marker character and get suggested values for the text matches so that you don't have to write it on your own.
+	 *
+	 * The recommended way to add mentions feature to an editor is by
+	 * using {@link CKEDITOR.config#mentions config.mentions} property or by passing it as the configuration option when instantiating an editor.
+	 *
+	 * ```javascript
+	 * // Simple usage with CKEDITOR.config.mentions property.
+	 * CKEDITOR.config.mentions = [ { feed: ['Anna', 'Thomas', 'John'], minChars: 0 } ];
```

Comment:
Recommended way is to set it up by passing the config directly to the `editor` instance, global config, although valid, is not a common method in CKE4. So I'd switch these two.